TYPO3.Flow/Classes/TYPO3/Flow/Persistence/Doctrine/Mapping/Driver/FlowAnnotationDriver.php

```diff
@@ -203,9 +203,14 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
             if ($entityAnnotation->readOnly) {
                 $metadata->markReadOnly();
             }
-        } elseif ($classSchema->getModelType() === ClassSchema::MODELTYPE_VALUEOBJECT) {
-            // also ok... but we make it read-only
-            $metadata->markReadOnly();
+        } elseif (isset($classAnnotations[Flow\ValueObject::class])) {
+            $valueObjectAnnotation = $classAnnotations[Flow\ValueObject::class];
+            if ($valueObjectAnnotation->embedded === true) {
+                $metadata->isEmbeddedClass = true;
+            } else {
+                // also ok... but we make it read-only
+                $metadata->markReadOnly();
```

Comment:
I'm unsure, the original Doctrine Annotation Driver also only marks readOnly annotated `Entities`. It will effectively completely skip the changeSet computation and I don't know if Doctrine needs that for the Entities where the VO is embedded to work correctly.